app/services/ncs_service/plant/move.rb

```diff
@@ -21,31 +21,45 @@ def transaction
         @transaction ||= get_transaction(:move_batch, @attributes.merge(zone: zone))
       end
 
-      def prior_move_transactions
-        Transaction.where(
-          'batch_id = ? AND type = ? AND vendor = ? AND id NOT IN (?)',
-          @batch_id,
-          :move_batch,
-          :ncs,
-          transaction.id
-        )
-      end
-
       def zone
         @zone ||= batch&.zone&.attributes
       end
 
+      def prior_move
+        previous_move = batch.completions.select do |c|
+          c.action_type == 'move' && c.id < @completion_id
+        end.max_by(&:start_time)
+
+        return if previous_move.nil?
+
+        # calling get_completion here will ensure relationships are side loaded.
+        get_completion(previous_move&.id)
+      end
+      memoize
+
+      def prior_start
+        previous_start = batch.completions.select do |c|
+          c.action_type == 'start' && c.id < @completion_id
+        end.max_by(&:start_time)
+
+        return if previous_start.nil?
+
+        # calling get_completion here will ensure relationships are side loaded.
+        get_completion(previous_start&.id)
+      end
+      memoize
```

Comment:
you forgot the method name as a parameter for `memoize`

Reply:
✅